en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/message-transformation/interceptor-microservice/sample-interceptor-services.md

```diff
@@ -0,0 +1,100 @@
+# Sample Interceptor Services
+
+## Sample 1: JSON to XML transformation for a Legacy Backend - Book Store
+
+Let's say we have a legacy backend named "Book Store". The resource `/books` with the method `POST` accepts an XML
+request to add a new book. If the header `x-user: admin` is not present in the request to the server, it responds
+with the `401` HTTP status code. If the book insertion is successful, it is `200`.
+
+Let's say we require to make the API accepts JSON payload and if the book gets created success,
+respond to it with the HTTP status code `201`, and we are happy about the response returned from the backend so, no need to modify the response.
+
+The following diagram describes what we are planning to do as in our requirements.
+
+<img src="{{base_path}}/assets/img/deploy/mgw/interceptor-example-json-to-xml.png" alt="Interceptor sample JSON to XML" width="880px"/>
+
+Since we read the request body in request interceptor, we need to add `request_body` in the `includes` section of Open API Definition of the API.
```

Comment:
```suggestion
Since you read the request body in request interceptor, you need to add `request_body` in the `includes` section of Open API Definition of the API.
```